app/controllers/journeys/expenses/repairsandmaintenance/RepairsAndMaintenanceAmountController.scala

```diff
@@ -33,42 +38,64 @@ import scala.concurrent.{ExecutionContext, Future}
 
 class RepairsAndMaintenanceAmountController @Inject() (
     override val messagesApi: MessagesApi,
-    sessionRepository: SessionRepository,
+    selfEmploymentService: SelfEmploymentServiceBase,
     navigator: ExpensesNavigator,
     identify: IdentifierAction,
     getData: DataRetrievalAction,
-    requireData: DataRequiredAction,
     formProvider: RepairsAndMaintenanceAmountFormProvider,
     val controllerComponents: MessagesControllerComponents,
     view: RepairsAndMaintenanceAmountView
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController
     with I18nSupport {
 
-  private def form(implicit request: DataRequest[AnyContent]) = formProvider(request.user.userType)
+  def onPageLoad(taxYear: TaxYear, businessId: BusinessId, mode: Mode): Action[AnyContent] = (identify andThen getData).async { implicit request =>
+    val result = for {
+      accountingTypeStr <- EitherT(selfEmploymentService.getAccountingType(request.user.nino, businessId.value, request.user.mtditid))
+      accountingType = AccountingType.withName(accountingTypeStr.toUpperCase())
```

Comment:
Would be nicer for the service to parse the accounting type into a `AccountingType` instead of a String - might be out the scope of this ticket though as that would affect quite a few controllers. Just an observation we can do later on.

Reply:
oh yes, the service will change in future - right now I kept it with string as it affects too many controllers